lib/tessel/deploy.js

```diff
@@ -124,58 +114,70 @@ Tessel.prototype.memoryInfo = function() {
       .catch(reject);
   });
 };
-/*
-  Run a script on a Tessel
-    param opts: unused so far
-*/
-Tessel.prototype.deployScript = function(opts) {
+/**
+ * Deploy project to a Tessel 2.
+ * Language: *
+ *
+ * @return {Promise}
+ */
+Tessel.prototype.deploy = function(opts) {
   // Only an _explicit_ `true` will set push mode
   var isPush = opts.push === true;
   var entryPoint = opts.entryPoint;
 
+  // Resolve the application's language/runtime
+  opts.lang = deployment.resolveLanguage(opts.lang || opts.entryPoint);
+
   return new Promise((resolve, reject) => {
-    // Stop running any existing scripts
-    return actions.execRemoteCommand(this, 'stopRunningScript')
+    // Stop running an existing applications
+    return this.simpleExec(commands.app.stop())
+      .catch((error) => {
+        // This _must_ be inline
+        if (error.length > 0) {
+          throw new Error(`Remote command: ${commands.app.stop().join(' ')} failed.`);
+        }
+      })
```

Comment:
This was the one case that `execRemoteCommand` was handling slightly differently, so in the interest of reduced operational code, this case handling is in-lined.